pkg/forwarder/forwarder_health.go

```diff
@@ -106,6 +110,24 @@ func (fh *forwarderHealth) healthCheckLoop() {
 	}
 }
 
+//populates a map containing API Endpoints per API keys that belongs to the forwarderHealth struct
+func (fh *forwarderHealth) setKeysPerAPIEndpoints() {
+	for domain, apiKeys := range fh.keysPerDomains {
+		apiDomain := ""
+		re := regexp.MustCompile("datadoghq.[a-z]*")
+		if re.MatchString(domain) {
+			apiDomain = "https://api." + re.FindString(domain)
+		} else {
+			apiDomain = domain
+		}
+		if _, ok := fh.keysPerAPIEndpoint[apiDomain]; ok {
+			fh.keysPerAPIEndpoint[apiDomain] = append(fh.keysPerAPIEndpoint[apiDomain], apiKeys...)
+		} else {
+			fh.keysPerAPIEndpoint[apiDomain] = apiKeys
+		}
```

Comment:
You don't need to check `keysPerAPIEndpoint` you can just append to it:

This will works even if `keysPerAPIEndpoint` is empty:
```go
fh.keysPerAPIEndpoint[apiDomain] = append(fh.keysPerAPIEndpoint[apiDomain], apiKeys...)
```